apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableVersion.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+
+namespace Google.Cloud.Bigtable.V2
+{
+    /// <summary>
+    /// A version which uniquely identifies a cell within a column.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Note: when using ReadModifyWriteRow, modified columns automatically use a server version, which
+    /// is based on the current timestamp micros. For those columns, other reads and writes should
+    /// use <see cref="BigtableVersion"/> values constructed from DateTime values, as opposed to using
+    /// a custom versioning scheme with 64-bit values.
+    /// </para>
+    /// </remarks>
+    public struct BigtableVersion : IComparable, IComparable<BigtableVersion>, IEquatable<BigtableVersion>
+    {
+        private const long TicksPerMicro = 10;
+
+        /// <summary>
+        /// Creates a new <see cref="BigtableVersion"/> value from a 64-bit value.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note: when using ReadModifyWriteRow, modified columns automatically use a server version, which
+        /// is based on the current timestamp micros. For those columns, other reads and writes should
+        /// use <see cref="BigtableVersion"/> values constructed from DateTime values, as opposed to using
+        /// a custom versioning scheme with 64-bit values.
+        /// </para>
+        /// </remarks>
+        /// <param name="value">
+        /// The non-negative version value, or -1 to initialize from the microseconds of DateTime.UtcNow.
```

Comment:
Just to check, does the Bigtable team recommend using client-side clocks like this?

Reply:
Yes, we had a thread about this in the API proposal doc with Gary Elliot.